### PR TITLE
Import Clusters to ACM Hub including a separated test mark

### PR DESCRIPTION
**Pull request by jimdevops19**
Signed-off-by: Cem Erginbas <cerginba@cerginba.tlv.csb>
Changes that are required for the importing scenario.
Some of the changes:
 - Added a new mark of `acm_import` as it will be a good solution for skipping pytest auto-use calls when entering run-ci
 - Added a new config file `conf/ocsci/import_clusters_to_acm.yaml` that skips ocs deployment and configures import_clusters_to_acm=True
 - Bug fixes in `ocsci/ocs/acm/acm.py` file that were required for the execution of the UI automation successfully to import the clusters
 - in `ocs_ci/ocs/ui/base_ui.py`  added chrome option of: `--disable-dev-shm-usage` when we pass the import_clusters_to_acm configuration. In such a scenario, the kubeconfig needs to be copied and sent as a text via UI automation. Since kubeconfig is a large file, Selenium crashes while trying to send those keys.
 - In `test_acm.py` file - the calls for submariner installation are deleted. We only want to handle import clusters for now.

### CODE REVIEW

File: tests/ecosystem/deployment/test_acm.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 ####################################################################################################
 # This file is placeholder for calling import ACM as test, until full solution will be implimented #
 ####################################################################################################
 
 
-def test_import_acm(setup_acm_ui):
-    acm_obj = AcmAddClusters(setup_acm_ui)
+@acm_import
+def test_acm_import(setup_acm_ui):
+    AcmAddClusters(setup_acm_ui)
```
Comment (OdedViner) [inline tests/ecosystem/deployment/test_acm.py:12]: I dont understand this test file...

File: tests/ecosystem/deployment/test_acm.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
     import_clusters_with_acm()
```
Comment (dahorak) [inline tests/ecosystem/deployment/test_acm.py:13]: This looks little bit strange - because there is instantiated the class `AcmAddClusters` here, but didn't stored in any variable and then there is the call of `import_clusters_with_acm()` function, which instantiated the class `AcmAddClusters` again and uses it.

File: ocs_ci/ocs/acm/acm.py [hunk 3]
```diff
@@ -369,12 +366,12 @@ def import_clusters_with_acm():
...
-    kubeconfig_a = clusters_env.get("kubeconfig_cl_a_location")
-    kubeconfig_b = clusters_env.get("kubeconfig_cl_b_location")
+    kubeconfig_a = clusters_env.get("kubeconfig_location_c1")
+    kubeconfig_b = clusters_env.get("kubeconfig_location_c2")
     global cluster_name_a
-    cluster_name_a = clusters_env.get("cl_a_cluster_name")
+    cluster_name_a = clusters_env.get("cluster_name_1")
     global cluster_name_b
-    cluster_name_b = clusters_env.get("cl_b_cluster_name")
+    cluster_name_b = clusters_env.get("cluster_name_2")
```
Comment (dahorak) [inline ocs_ci/ocs/acm/acm.py:374]: This doesn't look much scalable, but I understand that it was not introduced in this PR, but was already in the previous code. Also the usage of `global` is not nice in this context I think, but the same - it is not part of this PR.

File: tests/ecosystem/deployment/test_acm.py (hunk 2 above)
Comment (jimdevops19) [reply tests/ecosystem/deployment/test_acm.py:13]: Yes, this was a forgotten line to delete since some of its methods used to be called here. We only need the `import_clusters_with_acm()` with the instance created there. Since it assigns the selenium driver to the obj.

File: ocs_ci/ocs/acm/acm.py (hunk 3 above)
Comment (jimdevops19) [reply ocs_ci/ocs/acm/acm.py:374]: I am not a big fan of `global` as well but I think we should address it as a follow-up

File: ocs_ci/ocs/acm/acm.py [hunk 4]
```diff
@@ -77,6 +76,7 @@ def import_cluster(self, cluster_name, kubeconfig_location):
         ):
             if sample:
                 log.info(f"Cluster: {cluster_name} successfully imported")
+                return True
```
Comment (annagitel) [inline ocs_ci/ocs/acm/acm.py:79]: if you added  `return True` in the function, could you add `return False` after it? just so the function will return bool argument

Comment (annagitel) [reply ocs_ci/ocs/acm/acm.py:79]: and please add `Returns: ` section in the docs

File: ocs_ci/ocs/acm/acm.py [hunk 5]
```diff
@@ -332,30 +334,25 @@ def validate_cluster_import(cluster_name):
             dict_status.get("status") == "True"
         ), f"Status is not True, but: {dict_status.get('status')}"
 
+        # Return true if Assertion error was not raised:
+        return True
```
Comment (annagitel) [inline ocs_ci/ocs/acm/acm.py:338]: same as previous comment -True/False

File: tests/ecosystem/deployment/test_acm.py (hunk 1 above)
Comment (jimdevops19) [reply tests/ecosystem/deployment/test_acm.py:12]: This is the file for the ACM testing, we will include most likely functions that you want to execute after ACM is installed. Since it is inside the deployment parent directory, the functions here will execute deployment alike code, like the import scenario that is covered in this PR.

File: ocs_ci/ocs/acm/acm.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jimdevops19) [reply ocs_ci/ocs/acm/acm.py:338]: Thanks, missed this

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 6]
```diff
@@ -357,6 +356,11 @@ def pytest_configure(config):
                     "cluster is performed."
                 )
                 continue
+            elif markers_arg == "acm_import":
+                log.info(
+                    "Skipping auto pytest executions and version collecting because "
+                    "Import Clusters to ACM is performed."
+                )
```
[the remark this answers is not in the document]
Comment (jimdevops19) [reply ocs_ci/framework/pytest_customization/ocscilib.py:363]: Yes, I agree I should try that.

File: ocs_ci/ocs/acm/acm.py (hunk 4 above)
Comment (jimdevops19) [reply ocs_ci/ocs/acm/acm.py:79]: Good point. The goal is to get out of the loop once the sampler is not None so I'll try `return` first.

**Review by petr-balogh — APPROVED** [review]

**Comment by petr-balogh** [comment]
@jimdevops19 please open issue for documentation.

**Comment by jimdevops19** [comment]
> @jimdevops19 please open issue for documentation.

Done

**Review by OdedViner — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]
